### PR TITLE
Arkitect support

### PR DESCRIPTION
**Pull request by JTubex**
| Q             | A
| ------------- | ---
| Branch        | master for features and deprecations
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Documented?   | yes
| Fixed tickets | #1036

# New Task Checklist:

- [x] Are the dependencies added to the composer.json suggestions?
- [x] Is the doc/tasks.md file updated?
- [x] Are the task parameters documented?
- [x] Is the task registered in the tasks.yml file?
- [x] Does the task contains phpunit tests?
- [ ] Is the configuration having logical allowed types?
- [x] Does the task run in the correct context?
- [x] Is the `run()` method readable?
- [x] Is the `run()` method using the configuration correctly?
- [x] Are all CI services returning green?

### CODE REVIEW

File: src/Task/PhpArkitect.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'config' => null,
+            "target_php_version" => null,
+            'stop_on_failure' => null,
+        ]);
+
+        $resolver->addAllowedTypes('config', ['null', 'string']);
+        $resolver->addAllowedTypes('target_php_version', ['null', 'string']);
+        $resolver->addAllowedTypes('stop_on_failure', ['null', 'boolean']);
```
Comment (veewee) [inline src/Task/PhpArkitect.php:30]: Maybe better to default to false and only accept bools?

File: src/Task/PhpArkitect.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    public function run(ContextInterface $context): TaskResultInterface
+    {
+        $config = $this->getConfig()->getOptions();
+
+        $arguments = $this->processBuilder->createArgumentsForCommand('phparkitect');
+        $arguments->add('check');
+        $arguments->addOptionalArgument('--config=%s', $config['config']);
+        $arguments->addOptionalArgument('--target-php-version=%s', $config['target_php_version']);
+        $arguments->addOptionalArgument('--stop-on-failure', $config['stop_on_failure']);
+        $arguments->add('--no-ansi');
```
Comment (veewee) [inline src/Task/PhpArkitect.php:55]: Would it make sense not to provide this argument and let the system figure out if it needs to use ansi or not?
(the tool is based on symfony/console, so it should be able to detect and display ansi as wel)

**Review by veewee — COMMENTED** [review]
Thanks for the PR. Added some small things to the review.

File: src/Task/PhpArkitect.php (hunk 2 above)
Comment (JTubex) [reply src/Task/PhpArkitect.php:55]: I based this class on the PhpStan task. The `--no-ansi` is added there by default. I wasn't aware symfony/console detects this.

**Comment by veewee** [comment]
Nice! Thanks.

**Comment by JTubex** [comment]
I was checking the MR again and forgot to update the default value in the docs 🙈

Created #1039

**Pull request merged**